### PR TITLE
simplify PR template

### PR DESCRIPTION
**Pull request by geekflyer**
### Description

This PR simplifies the PR template and cleans up unnecessary stuff.
I noticed we have a good amount of PRs which still contain a lot of boilerplate from the original template which just adds noise.
With this change the hope is we get more meaningful and dense PR descriptions.
A typical PR simply should have a Description (that is implied, so no header required for that.) and a Test Plan.
Related PRs can and should be mentioned in context of the PR Description.

### Test Plan
not necessary

### CODE REVIEW

**Review by davidiw — CHANGES_REQUESTED** [review]
While I agree with the goal -- I have often wanted to just trim this as well. We should explicitly have motivation / description and test plan. We also need CLA until we are fortunate enough to remove it.

I think it isn't bad to set expectations.

**Comment by geekflyer** [comment]
ok, I added a `### Description` back to the template although I think this is already kinda implied due to the overall PR field being called description.
For the CLA discussion I'll just wait a few days until this is cleared by legal to be removed before merging this PR

**Comment by davidiw** [comment]
> ok, I added a `### Description` back to the template although I think this is already kinda implied (due to the overall PR field being called description) and pretty obvious that a PR needs a description. For the CLA discussion I'll just wait a few days until this is cleared by legal to be removed before merging this PR

I like your optimism :)

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (JoshLind) [inline .github/PULL_REQUEST_TEMPLATE.md:1]: @geekflyer, if I'm being picky, I'd like the template to have:
1. The header  `<!-- .... -->`. I think it's good to have this context and intro 😄.
2. The describing text for both sections, e.g.: `(Describe the proposed change here and the motivation for the change.)` and `(Share your test plan here. Please provide us with clear details for verifying that your changes work.)`

Having these here forces the author to think about them and allows reviewers to reject sloppy work.

**Review by JoshLind — COMMENTED** [review]
@geekflyer, no more CLA 😄 Happy to approve if we can find the right balance.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (geekflyer) [reply .github/PULL_REQUEST_TEMPLATE.md:1]: @JoshLind 

I honestly would prefer to have it really minimalist and only add what's relevant context required for 90+% of PRs.
The reasons for that are:
- remove noise from PRs where people forget to remove a section or just enter some dummy placeholder text (just a recent example: https://github.com/aptos-labs/aptos-core/pull/1316#issuecomment-1150445445 )
- reduce the number of keystrokes/copy/delete/select operations an author has to do to create a PR. I really like it to be intuitive such that when creating a PR I can enter a description right away.
- people are busy. The more noise/boilerplate there is a PR template, the less likely they're going to actually reading/thinking about the _important_ instructions in a PR template (like the fact that somewhere at the bottom we request a test plan).

With that in mind, I think:
1.  
> Thank you for sending a PR. We appreciate you spending time to help improve the Aptos Core project.

is pretty useless for most PRs (especially our own). I think it would be much more genuine and meaningful if we just say thank you in the PR review comment (which we do). If we still want to have have some kind of "automated thank you" we can add a bot or add it to to the contributing.md but I don't think it belongs into the PR template.

2. The text describing both sections
> Describe the proposed change here and the motivation for the change.
I think that's pretty obvious. Any good PR should describe the proposed change and motivation. Unfortunately I haven't seen an indication that having this placeholder text there actually _forces_ and author to do so or even increases the odds at having a PR good with a _good_ description. The only thing that truly impacts PR description quality is the review culture + reviewer stringency.

3.
> Share your test plan here. Please provide us with clear details for verifying that your changes work.
The `Share your test plan here. ` is pretty obvious from the Heading title, so we can remove that.
I think I'm fine keeping the `Please provide us with clear details for verifying that your changes work` just to make it a bit more explicit that we care about this. I'd probably wrap it in a ""  which make it hidden once submitted.

With that in mind the final PR template would look like this:
```
### Description

### Test Plan
<!-- Please provide us with clear details for verifying that your changes work. -->
```

Is that something agreeable? 😺 

Also btw, we can add more stuff back over time if we feel the need, but I'd like to start from a minimalist template and only add things as truly required.

Comment (JoshLind) [reply .github/PULL_REQUEST_TEMPLATE.md:1]: @geekflyer, I still have mixed feelings. 😄 In my mind, this description is for PR authors who are not part of the Aptos team, so I'm less worried about our PRs not following these instructions to the letter. However, I'll concede that having no CLA is a win. If I frequently find myself having to ask developers to add a description when I'm reviewing, I'll update the template. Unblocking this for now!

**Review by JoshLind — APPROVED** [review]
Unblock for no CLA! 🥳

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (geekflyer) [reply .github/PULL_REQUEST_TEMPLATE.md:1]: > Unblock for no CLA! 🥳

ok cool, let's give it a shot and if we see that the PR quality deteriorates  we can add some more explicit instructions back to the template :)

**Pull request closed without merging**
